src/actionSelector.ts

```diff
@@ -14,10 +14,12 @@ Big.DP = 30;
 export class ActionSelector {
   private config: Config;
   private selectedActions: Action[];
+  private rates: Rates;
 
-  constructor(config: Config) {
+  constructor(config: Config, rates: Rates) {
```

Comment:
🍌 Just a thought experiment: for me the `ActionSelector` is stupid in a way that it only _selects_ the action from the passed in `Entity`. However, in case of `Accept` and `Decline` this would mean we would need to return both and the `ActionExecutor` needs to decide which action to execute. 
I'm thinking in this direction, because I have a hard time to draw a line between the `ActionSelector` and `ActionExecutor` and who does what.

Reply:
yes, but that is beyond this PR 😃